### PR TITLE
89571 New SRMA live page (SRMA story 4)

### PR DESCRIPTION
**Pull request by Menol-LK**
**What is the change?**
New SRMA live page (SRMA story 4)

**Why do we need the change?**
New feature

**What is the impact?**
New feature - Minimum impact on existing components

**Azure DevOps Ticket**
89571/91247

### CODE REVIEW

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/SRMA/Index.cshtml.cs [hunk 1]
```diff
@@ -1,44 +1,58 @@
...
+		public IndexPageModel(ISRMAService srmaService, ILogger<IndexPageModel> logger)
 		{
+			_srmaService = srmaService;
 			_logger = logger;
 		}
 
 		public async Task OnGetAsync()
 		{
-			try
+			if (long.TryParse(Convert.ToString(RouteData.Values["urn"]), out long caseUrn) == false || caseUrn <= 0)
```
Comment (elielijah321) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/SRMA/Index.cshtml.cs:31]: Can we make this check similar to this `(caseUrnValue == null || !long.TryParse(caseUrnValue.ToString(), out long caseUrn) || caseUrn == 0)` for the sake of consistency

Comment (elielijah321) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/SRMA/Index.cshtml.cs:31]: `~\Pages\Case\Management\Concern\Closure.cshtml.cs` GetRouteData has an example of a cleaner way to check route data

File: ConcernsCaseWork/ConcernsCaseWork/Models/CaseActions/SRMA.cs [hunk 2]
```diff
@@ -5,8 +5,15 @@ namespace ConcernsCaseWork.Models.CaseActions
 {
 	public class SRMA : CaseAction
```
Comment (elielijah321) [inline ConcernsCaseWork/ConcernsCaseWork/Models/CaseActions/SRMA.cs:6]: Could you change the name of this class to SRMAModel

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/SRMA/Index.cshtml [hunk 3]
```diff
@@ -1,28 +1,231 @@
-﻿@page "/case/{urn:long}/management/action/srma"
-@model ConcernsCaseWork.Pages.Case.Management.Action.Srma.IndexPageModel
+﻿@page "/case/{urn:long}/management/action/srma/{srmaId:long}"
+@model ConcernsCaseWork.Pages.Case.Management.Action.SRMA.IndexPageModel
```
Comment (elielijah321) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/SRMA/Index.cshtml:2]: Is there a reason why you've reverted the Srma namespace back to SRMA?

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/SRMA/Index.cshtml.cs [hunk 4]
```diff
@@ -1,44 +1,58 @@
...
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Logging;
 using System;
+using System.Linq;
 using System.Threading.Tasks;
+using CaseActionModels = ConcernsCaseWork.Models.CaseActions;
 
-namespace ConcernsCaseWork.Pages.Case.Management.Action.Srma
+namespace ConcernsCaseWork.Pages.Case.Management.Action.SRMA
```
Comment (elielijah321) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/SRMA/Index.cshtml.cs:11]: Namespace has been reverted here

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/SRMA/Index.cshtml.cs (hunk 1 above)
Comment (Menol-LK) [reply ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/SRMA/Index.cshtml.cs]: pushed the suggested change

File: ConcernsCaseWork/ConcernsCaseWork/Models/CaseActions/SRMA.cs (hunk 2 above)
Comment (Menol-LK) [reply ConcernsCaseWork/ConcernsCaseWork/Models/CaseActions/SRMA.cs]: pushed the suggested change

**Review by elielijah321 — APPROVED** [review]

**Pull request merged**